lightningd/peer_htlcs.c

```diff
@@ -642,6 +643,26 @@ htlc_accepted_hook_deserialize(const tal_t *ctx, const char *buffer,
 static void htlc_accepted_hook_serialize(struct htlc_accepted_hook_payload *p,
 					 struct json_stream *s)
 {
+	const struct route_step *rs = p->route_step;
+	const struct htlc_in *hin = p->hin;
+	json_object_start(s, "onion");
+
+	json_object_start(s, "hop_data");
```

Comment:
I think we need a hop_data_raw field, which is the literal hex, which is always present. Then this object should be called "realm0" and only present if realm is 0.  That makes the API more future-proof.

Reply:
Ah, I see this in the later patch.  We should use spec naming for our APIs, so this should be called "per_hop" and probably "per_hop0" for realm 0.

Reply:
Changed name to `per_hop_v0` and added an `if` statement around this object that will cause a compilation error once we merge PR #2689 due to it referencing the `hop_data->realm`.